### PR TITLE
Update GitHub Actions

### PR DESCRIPTION
**Pull request by smorimoto**
https://github.com/smorimoto/ocaml-uri/actions/runs/10027492512

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -11,33 +11,21 @@ jobs:
...
-      - name: Tests
-        run: |
-          opam exec -- dune build @runtest
-
-      - name: Opam Lint
-        run: |
-          opam lint uri.opam uri-re.opam uri-sexp.opam uri-bench.opam
+      - run: opam install . --deps-only --with-test
+      - run: opam exec -- dune build
+      - run: opam exec -- dune build @runtest
```
Comment (samoht) [inline .github/workflows/test.yml:31]: @smorimoto could you also re-add the `lint` step that you have removed in your patch?

File: .github/dependabot.yml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+version: 2
```
Comment (samoht) [inline .github/dependabot.yml:1]: Is it useful to run dependabot on this repo?

File: .github/workflows/test.yml (hunk 1 above)
Comment (smorimoto) [reply .github/workflows/test.yml:31]: Oops, sure!

**Comment by smorimoto** [comment]
By specifying major versions 4 and 5 for the OCaml version, you can test with the latest major releases. This means that by testing the minimum requirement, 4.08, and the latest version of each major version, you can theoretically check loosely that all versions work.

**Comment by samoht** [comment]
Many thanks!

**Pull request merged**